packages/platform-sdk-lsk/src/services/ledger.ts

```diff
@@ -80,26 +79,21 @@ export class LedgerService extends Services.AbstractLedgerService {
 
 		const addresses: string[] = [];
 
-		const path = `m/44'/${slip44}'/0'`;
 		let initialAddressIndex = 0;
 
 		if (options?.startPath) {
 			initialAddressIndex = BIP44.parse(options.startPath).addressIndex + 1;
 		}
 
-		const compressedPublicKey = await this.getExtendedPublicKey(path);
-
 		for (const addressIndexIterator of createRange(page, pageSize)) {
 			const addressIndex = initialAddressIndex + addressIndexIterator;
-			const publicKey: string = HDKey.fromCompressedPublicKey(compressedPublicKey)
-				.derive(`m/0/${addressIndex}`)
-				.publicKey.toString("hex");
-
-			const { address } = await this.#identity.address().fromPublicKey(publicKey);
 
+			const path = `m/44'/${slip44}'/0'/0/${addressIndex}`;
+			const { publicKey, address } = await this.getPublicKeyAndAddress(path);
+			console.log("path", path, "publicKey", publicKey, "address", address);
```

Comment:
@faustbrian I finally got this to work. But apparently LSK, just like ARK, doesn't have multiple addresses for same account. So in this code, we're hard-coding account id to `0` (which we should probably allow for iterating over it to have more wallets, but this is something to resolve with all coins) and then even if we iterate over the address index, nano is returning always the same pubkey and address. This is the sample output.
```
    path m/44'/134'/0'/0/0 publicKey d48522677df50defd175c85072309c7643dbc6bdc63c7665a302579ed2ccaedb address 7399986239080551550L
    path m/44'/134'/0'/0/1 publicKey d48522677df50defd175c85072309c7643dbc6bdc63c7665a302579ed2ccaedb address 7399986239080551550L
    path m/44'/134'/0'/0/2 publicKey d48522677df50defd175c85072309c7643dbc6bdc63c7665a302579ed2ccaedb address 7399986239080551550L
    path m/44'/134'/0'/0/3 publicKey d48522677df50defd175c85072309c7643dbc6bdc63c7665a302579ed2ccaedb address 7399986239080551550L
    path m/44'/134'/0'/0/4 publicKey d48522677df50defd175c85072309c7643dbc6bdc63c7665a302579ed2ccaedb address 7399986239080551550L
```

Reply:
Looking at `getPublicKeyAndAddress` and `getLedgerAccount` it always only derives based on the account but ignores the change and address.

Reply:
Looking at how they derive wallets in https://github.com/LiskHQ/lisk-desktop I think we need to do 1 account = 1 wallet.